example/imaginary-wasm.hh

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include <functional>
+#include <initializer_list>
+#include <string>
+#include <vector>
+
+namespace wasm {
+struct Frame {
+  // TODO: what are the pointer invalidation rules here?
+  const Instance *instance() const;
+
+  size_t module_offset() const;
+  uint32_t func_index() const;
+  size_t func_offset() const;
+};
+
+struct Error : std::runtime_error {};
+
+// An error that occurs with a wasm function in the call stack.
+struct Trap : Error {
+  typedef iterator Frame*;
+  typedef const_iterator const Frame*;
+  Frame origin();
+  iterator begin();
+  iterator end();
+  const_iterator begin() const;
+  const_iterator end() const;
+};
+struct EngineConfig {
+  EngineConfig();
+  // Implementation provides extra c'tors here.
```

Comment:
How does this happen? By editing this file and shipping their own version of it?